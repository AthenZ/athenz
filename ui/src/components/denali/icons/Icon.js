/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import P, { PropTypes } from 'prop-types';
import { ICONS } from './Icons';
import { css } from 'emotion';

const Icon = (props) => {
    const icon = ICONS[props.icon];
    const width = props.size ? props.size : props.width;
    const height = props.size ? props.size : props.height;
    return (
        <svg
            viewBox='0 0 1024 1024'
            width={width}
            height={height}
            className={css`
                fill: ${props.color};
                cursor: ${props.isLink ? 'pointer' : 'inherit'};
                vertical-align: ${props.verticalAlign};
            `}
            onClick={props.onClick}
            ref={props.innerRef}
            data-testid='icon'
        >
            <title>{props.icon}</title>
            {icon.map((path, index) => (
                <path key={index} d={path} />
            ))}
        </svg>
    );
};

Icon.propTypes = {
    icon: PropTypes.string.isRequired,
    /* Can be any CSS size (eg. 16px, 1rem, 1.725em). Default: 1rem */
    size: PropTypes.string,
    /* can be used when size height and width are different */
    width: PropTypes.string,
    height: PropTypes.string,
    /* Can be any CSS color (eg. 'red', '#fff', rgba(128, 34, 64, 0.3)) */
    color: PropTypes.string,
    /* Default: text-bottom */
    verticalAlign: PropTypes.string,
    /* Click handler */
    onClick: PropTypes.func,
    isLink: PropTypes.bool,
    innerRef: P.oneOfType([P.func, P.object]),
};

Icon.defaultProps = {
    icon: 'x',
    size: '1em',
    verticalAlign: 'text-bottom',
};

export default Icon;
