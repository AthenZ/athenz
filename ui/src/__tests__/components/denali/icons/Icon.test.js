/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import Icon from '../../../../components/denali/icons/Icon';
import { render } from '@testing-library/react';

describe('Icon', () => {
    it('should render default icon', () => {
        let icon = <Icon />;
        expect(icon.props.icon).toEqual('x');
        expect(icon.props.size).toEqual('1em');
        expect(icon.props.verticalAlign).toEqual('text-bottom');
    });

    it('should render an icon with values provided', () => {
        let icon = <Icon icon='accountkey' size='1em' color='icons-white' />;
        expect(icon.props.icon).toEqual('accountkey');
        expect(icon.props.size).toEqual('1em');
        expect(icon.props.color).toEqual('icons-white');
    });

    it('should render default icon with svg', () => {
        const { getByTestId } = render(<Icon />);
        const icon = getByTestId('icon');
        expect(icon).toMatchSnapshot();
    });
});
