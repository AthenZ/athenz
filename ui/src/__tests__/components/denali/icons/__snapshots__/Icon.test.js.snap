// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render default icon with svg 1`] = `
.emotion-0 {
  cursor: inherit;
  vertical-align: text-bottom;
}

<svg
  class="emotion-0"
  data-testid="icon"
  height="1em"
  viewBox="0 0 1024 1024"
  width="1em"
>
  <title>
    x
  </title>
  <path
    d="M572.373 512l224.427-224.213c8.506-8.066 13.798-19.448 13.798-32.066 0-9.118-2.763-17.59-7.498-24.625l0.1 0.158c-7.858-10.565-20.305-17.335-34.333-17.335-11.534 0-21.999 4.577-29.678 12.013l0.011-0.011-227.2 225.707-226.133-225.707c-7.733-7.792-18.447-12.616-30.289-12.616-14.295 0-26.948 7.030-34.691 17.822l-0.087 0.128c-4.635 6.877-7.398 15.35-7.398 24.467 0 12.618 5.292 24 13.778 32.048l0.020 0.019 224.427 224.213-224.427 224.213c-8.506 8.066-13.798 19.448-13.798 32.066 0 9.118 2.763 17.59 7.498 24.625l-0.1-0.158c7.83 10.919 20.483 17.949 34.778 17.949 11.841 0 22.556-4.824 30.286-12.613l0.003-0.003 226.133-225.707 226.133 225.707c7.668 7.425 18.133 12.002 29.667 12.002 14.028 0 26.476-6.77 34.252-17.222l0.081-0.114c4.635-6.877 7.398-15.35 7.398-24.467 0-12.618-5.292-24-13.778-32.048l-0.020-0.019z"
  />
</svg>
`;
